s3-zip-store/src/main/scala/zip_partitioner/s3/S3ZipStore.scala

```diff
@@ -0,0 +1,75 @@
+package zip_partitioner.s3
+
+import cats.effect.{IO, Resource}
+import eu.timepit.refined.types.string.NonEmptyString
+import fs2.Pipe
+import fs2.aws.s3.S3
+import fs2.aws.s3.models.Models.{BucketName, ETag, FileKey}
+import fs2.compression.{DeflateParams, InflateParams, ZLibParams}
+import io.laserdisc.pure.s3.tagless.Interpreter
+import software.amazon.awssdk.services.s3.S3AsyncClient
+
+import java.util.zip.{ZipEntry, ZipOutputStream}
+
+object S3ZipStore {
+
+  def transfer(from: NonEmptyString, to: NonEmptyString, fileIdentifier: NonEmptyString, s3AsyncClient: S3AsyncClient): fs2.Stream[IO, ETag] =
+    withS3(s3AsyncClient).flatMap { s3 =>
+      val body = s3.readFile(BucketName(from), FileKey(fileIdentifier))
+      body.through(deflateStream)
+        .through(s3.uploadFile(BucketName(to), FileKey(fileIdentifier)))
+    }
+
+  def retrieveSingle(compressedBucket: NonEmptyString, fileKey: NonEmptyString, s3AsyncClient: S3AsyncClient): fs2.Stream[IO, Byte] =
+    withS3(s3AsyncClient).flatMap { s3 =>
+      val body = s3.readFile(BucketName(compressedBucket), FileKey(fileKey))
+      body.through(inflateStream)
+    }
+
+  // create a zip file in the form of a byte stream out of several file keys retrieved from s3
+  def retrieveMultiple(compressedBucket: NonEmptyString, fileKeys: List[NonEmptyString], streamSize: Int = 1024*512, s3AsyncClient: S3AsyncClient): fs2.Stream[IO, Byte] =
```

Comment:
should have the streamSize with default value last. but I think the best way here is to overload the method and have 
```scala
def retrieveMultiple(compressedBucket: NonEmptyString, fileKeys: List[NonEmptyString], s3AsyncClient: S3AsyncClient) =
    retrieveMultiple(compressedBucket, fileKeys, DefaultBufferSize, s3AsyncClient)
```